javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js

```diff
@@ -1,272 +1,259 @@
+ */
 
-  updateStatsUI() {
-    /**
-     * After keeping a set of dice in the 6th round, the current round text changes to 7 just before the game over confirmation message pops up.
-     * This change prevents updating that text without messing up the other logic that relies on the current round being greater that 6
-     * if (this.currentRound <= 6) currentRoundText.textContent = this.currentRound;
-     *
-     * We can point out this error to campers and create a step where campers add that if statement
-     *
-     *
-     */
-    currentRoundText.textContent = this.currentRound;
-    totalScoreText.textContent = this.totalScore;
-    currentRoundRollsText.textContent = this.rollsInCurrentRound;
-  }
+const straightDetector = (arr) => {
+  const sortNumbers = arr.sort((a, b) => a - b);
+  const uniqueNumbers = [...new Set(sortNumbers)];
+  const stringifyArray = uniqueNumbers.join("");
 
-  updateScoreUI() {
-    const scoreboard = scoreHistory;
-    const scoreEntry = document.createElement("li");
-    // I believe this is the first time Object.entries has been introduced
-    const [type, score] = Object.entries(
-      game.scoreHistory[game.scoreHistory.length - 1]
-    )[0];
-
-    const formattedScoreType = (type[0].toUpperCase() + type.slice(1)).replace(
-      /-/g,
-      " "
-    );
-
-    const formattedScore = document.createTextNode(
-      `${formattedScoreType}: ${score}`
-    );
-    scoreEntry.appendChild(formattedScore);
-    scoreboard.appendChild(scoreEntry);
-  }
+  // SMALL STRAIGHT POSSIBILITIES
+  // 1,2,3,4
+  // 2,3,4,5
+  // 3,4,5,6
 
-  generateValidScoreOptions() {
-    const sortedDiceValues = selectionSort(this.diceValues);
-    const sumOfDiceValues = sumOfArrElements(this.diceValues);
-    const totalConsecutives = getMaxNumOfConsecutives(sortedDiceValues);
-
-    // I believe this is the first time Object.values has been introduced
-    const numOfSameDiceValues = Object.values(
-      getNumOfSameDiceValues(sortedDiceValues)
-    );
-
-    this.validScoreOptions = {
-      none: 0,
-    };
-
-    for (let i = 0; i < numOfSameDiceValues.length; i++) {
-      if (numOfSameDiceValues[i] >= 3) {
-        this.validScoreOptions["three-of-a-kind"] = sumOfDiceValues;
-        break;
-      }
-    }
+  // LARGE STRAIGHT POSSIBILITIES
+  // 1,2,3,4,5
+  // 2,3,4,5,6
 
-    if (numOfSameDiceValues[0] >= 4)
-      this.validScoreOptions["four-of-a-kind"] = sumOfDiceValues;
+  const smallStraightLogic = ["1234", "2345", "3456"];
+  const largeStraightLogic = ["12345", "23456"];
 
-    if (numOfSameDiceValues.length === 2) {
-      if (sumOfArrElements(numOfSameDiceValues) === 5) {
-        this.validScoreOptions["full-house"] = 25;
-      }
-    }
+  if (smallStraightLogic.includes(stringifyArray)) {
+    updateRadioOption(3, 30);
+  } else if (largeStraightLogic.includes(stringifyArray)) {
+    updateRadioOption(4, 40);
+  } else {
+    updateRadioOption(5, 0);
+  }
```

Comment:
```suggestion
  if (smallStraightLogic.includes(stringifyArray)) {
    updateRadioOption(3, 30);
  }
  if (largeStraightLogic.includes(stringifyArray)) {
    updateRadioOption(4, 40);
  }
  updateRadioOption(5, 0);
```

These should not be mutually exclusive, IMHO.